virtual_rainforest/core/constants.py

```diff
@@ -0,0 +1,132 @@
+"""The :mod:`~virtual_rainforest.core.constants` module is used to store constants that
+are used across the Virtual Rainforest. This includes universal constants, such as the
+strength of gravity, but also constants that are shared between multiple models, such as
+the depth of the biogeochemically active soil layer.
+
+At the moment, no constants are actually stored in this module. It currently only
+contains the :attr:`CONSTANTS_REGISTRY`, which all constants classes should be
+registered in across models. This allows for all constants to be documented neatly.
+"""  # noqa: D205, D415
+
+import dataclasses
+from importlib import import_module
+from typing import Any, Callable
+
+from virtual_rainforest.core.exceptions import ConfigurationError
+from virtual_rainforest.core.logger import LOGGER
+
+CONSTANTS_REGISTRY: dict[str, dict[str, Callable]] = {}
+"""A registry for all the constants data classes.
+
+:meta hide-value:
+"""
+
+
+def register_constants_class(model_name: str, class_name: str) -> None:
```

Comment:
Could users supply the class type to this function directly? E.g.:

```py
register_constants_class("my_model", MyConstants)
```

You can get a class's name from the `__name__` attribute.

That would mean you could get rid of the `import_module` shenanigans.